spec/controllers/users_controller_spec.rb

```diff
@@ -160,4 +160,15 @@
       expect(user.locale).to eq('es')
     end
   end
+
+  describe '#show' do
+    context 'when user not found' do
+      before {allow(controller).to receive(:current_user).and_return(nil) }
+      it 'redirects to the home page' do
+       @controller = DashboardController.new
```

Comment:
This means you are using the DashboardController rather than the UsersController. This shouldn't be necessary here.